### DISCUSSION TITLE
Welcome to Microsoft Power Fx Discussions!

### DISCUSSION THREAD

**Discussion by gregli-msft** [microsoft/Power-Fx#2] [General]
## Welcome to Microsoft Power Fx!
  We’re using Discussions as a place to connect with other members of our community. We hope that you:
  * Ask questions you’re wondering about.
  * Share ideas.
  * Engage with other community members.
  * Welcome others and are open-minded. Remember that this is a community we
  build together.

**Comment by marsanyi** [comment]
This is really interesting to me.
Years ago, I worked in experimental electronic music (don't laugh).  I had an idea for a language that expressed a given piece as a network of musical entities, each consisting of a bundle of properties (like "loudness", "density", ...) that themselves were expressed in terms of other entity properties.  Some properties were tied to objective attributes (like "current time", "last note played", ...)  Playing the piece would consist of running the network and interacting with it.
Power FX looks like the language to express that: declarative, where entities are defined in terms of other entities and their properties, updated in real time.  Of course, the set of things that the language knows about currently is relevant more to business users than musicians.  How hard is it to add entities to the language that implement side-effects of the functional declarations in the same way as, for example, the GUI widgets (for example, a MIDI input and output)?

**Comment by orcmid** [comment]
My name is Dennis E. Hamilton. My Internet and gaming handle is "orcmid." I have had that since around 1978 when I participated in my first on-line multiplayer game, DecWars.

May 2021 will be the 63rd anniversary of my first computer program (in the original Fortran).  It halted the IBM 704 at Boeing with a divide check.  So much for using textbook formulas that are unstable at roots of a function :).

My first deep fascination with programming languages came on encountering the Report on ALGOL 60 in Communications of the ACM.  I was inspired and used to write BNF like my high-school buddies and I liked to draw model-train layouts.

I have managed to work on a couple of implemented programming languages and I am currently interested in a small language that demonstrates a model of computation that I trust will invite understanding of the stored-program concept and the power of digital representations.  It doesn't make representation easy, but that is sort of the point.

Document Engineering has become a passion of mine, as well as document processing.  I knew Don Knuth before TAOCP, and we had a shared interest and I saw some of his pre-TeX ideas.  (preTeX; hmm, I should save that one.)   I went from document management to document formats (ODA and then OOXML and, umm, ODF) including some attention to OpenFormula.  By Document Engineering I mean the counterpart of Software Engineering with regard to situated document creation and lifecycle activities.

I am also interested in digital audio and am blown-away by the capability of tools and plugins these days.  I also fancy cinematic quality computer games of the Tomb Raider and Ghost Recon, Division 2 flavors.  I have a project to translate the original Colossal Cave Adventure to modern/safe C as a demonstration, after getting permission to make a derivative work.  I got stuck in a tooling problem, and my axe is now very dull.  It hadn't occurred to me until this very moment to wonder how Power FX might host such a thing. 

Power FX interests me because of its declarative nature and the idea of having the Excel solver behind it.  And I am a big fan of grammar-early programming-language development.
